fontstash/fontstash.h

```diff
@@ -1567,7 +1615,13 @@ static __inline void fons__vertices(FONScontext* stash, FONSquad q, FONSstate* s
     fons__vertex(stash, q.x1, q.y1, q.s1, q.t1, state->color);
 }
 
-void fonsSetShaping(FONScontext* stash, const char* script, const char* direction, const char* language)
+void fonsInitShapingPlan(FONScontext* stash, const char* str) {
+    FONSUTF32string fons32String = stash->params.toUTF32Alloc(stash->params.userPtr, str);
+    fons__hb_initShapingPlan(stash, fons32String);
+    stash->params.UTF32Free(stash->params.userPtr, fons32String);
+}
```

Comment:
Yeah the lack of documentation (well, except reading through the mailing list or reading harfbuzz codebase) didn't help on that. What does hb_buffer_guess_segment_properties does exactly?
I think hb_buffer_get_glyph_infos can only be used after the shaping has been runned though?